### PR TITLE
chore: harden validation, security, and tests

### PR DESCRIPTION
**Pull request by marcelosalloum**
## What
- Reject non-address Soroban auth entries (was silently skipping them)
- Reject zero amounts and leading zeros in `validateAmount` (`/^[1-9]\d*$/`)
- Defer challenge nonce burn to after successful verification (both servers)
- Check `sendTransaction` status at all 5 broadcast sites before polling
- Reject `type="hash"` credentials when `feePayer=true` (spec §5.4)
- Enforce exactly one `invokeHostFunction` in push-mode (hash path) verification
- Reject multiple decimal points in `toBaseUnits`, handle `decimals=0` in `fromBaseUnits`
- Propagate RPC errors from `readCloseEffectiveAtLedger` instead of swallowing
- Clean up AbortSignal listener in Watcher `stop()`
- Replace magic number `5` with `DEFAULT_LEDGER_CLOSE_TIME` in client sponsored path
- Export all 4 error types from root (`StellarMppError`, `PaymentVerificationError`, `ChannelVerificationError`, `SettlementError`)
- Echo `externalId` in receipts for both charge and channel servers
- Add `Store.memory()` to charge server example for replay protection
- Fix `demo/run.sh` hardcoded port and missing `SERVER_URL`
- Update README: Node.js 22+ → 20+, complete exports table, add error types
- Update CLAUDE.md: peer dep versions, test count, SettlementError in module map
- Bump `package.json`: `engines >=20`, `sideEffects: false`, `homepage`, peer deps `^`
- Add 30 new tests (226 → 256): `close()` function, DUPLICATE status, exponential backoff, charge client/server behavior tests, constants exact values

## Why
Systematic review against `draft-stellar-charge-00` spec and production hardening audit. Closes gaps in input validation, replay protection, spec compliance, error handling, and test coverage. Every broadcast site now checks for ERROR/DUPLICATE before polling, challenge nonces are only consumed after successful verification (preventing permanent burn on transient failures), and push-mode verification now enforces single-operation transactions to prevent piggybacked operations.

## E2E verification
All 5 checks passed on Stellar testnet before commit:
- `make check` — 256 tests, 0 lint errors, types/build/format clean
- Example script validation — all 6 scripts load correctly
- Charge E2E (`demo/run.sh`) — 402 → sign → broadcast → **200 OK**
- Channel E2E (`demo/run-channel.sh`) — 2 requests, cumulative 1M → 2M stroops, both **200 OK**
- Channel E2E settlement (`demo/run-channel-e2e.sh`) — deploy → 2 payments → close → **balance: 0**

### CODE REVIEW

File: demo/run.sh [hunk 1]
```diff
@@ -56,7 +56,7 @@ trap "kill $SERVER_PID 2>/dev/null" EXIT
 
 # Wait for server to be ready
 for i in $(seq 1 10); do
-  if curl -s http://localhost:3000 >/dev/null 2>&1; then
+  if curl -s http://localhost:$PORT >/dev/null 2>&1; then
     break
```
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply demo/run.sh:60]: Addressed in b38ec9e
